src/bespokelabs/curator/request_processor/openai_batch_request_processor.py

```diff
@@ -721,20 +719,51 @@ async def submit_batch_from_request_file(
     async def track_already_submitted_batches(self):
         """
         Tracks previously submitted batches from the submitted batch objects file.
+        We need to check all submitted batch objects files because we might be looking at a cancelled batch
+        or a batch from another key but same project.
 
         Side Effects:
             - Updates tracker with previously submitted batch statuses
         """
-        if os.path.exists(self.submitted_batch_objects_file):
-            with open(self.submitted_batch_objects_file, "r") as f:
+        all_submitted_batches_files = set(
+            glob.glob(f"{self.working_dir}/batch_objects_submitted_*.jsonl")
+        )
+
+        existing_submitted_batches = {}
+        for submitted_batch_objects_file in all_submitted_batches_files:
+            logger.info(
+                f"Processing submitted batch objects file: {submitted_batch_objects_file} Your API key is ***{self.client.api_key[-4:]}."
+            )
+            with open(submitted_batch_objects_file, "r") as f:
                 for line in f:
                     batch_object = Batch.model_validate(json.loads(line))
                     request_file_name = batch_object.metadata["request_file_name"]
                     logger.debug(
                         f"Already submitted batch {batch_object.id} for request file {request_file_name}. "
                         f"Getting batch object to update tracker."
                     )
-                    batch_object = await self.retrieve_batch(batch_object.id)
+                    try:
+                        batch_object = await self.retrieve_batch(batch_object.id)
+                    except NotFoundError:
+                        logger.info(
+                            f"Batch {batch_object.id} not found. This is fine since we might be looking at a cancelled batch, "
+                            f"a deleted batch, or a batch from another project. Will continue..."
+                        )
+                        continue
+
+                    if not self._validate_batch_status(batch_object.status):
+                        logger.info(
```

Comment:
again warning.
and again, say that this batch was already submitted and will be resubmitted

Reply:
It might or might not be resubmitted as I explained above :)